### DISCUSSION TITLE
cannot install brownie with pipx

### DISCUSSION THREAD

**Discussion by s1n7** [smartcontractkit/full-blockchain-solidity-course-py#216] [Q&A] [answered]
Hello! 

I'm unable to install brownie with pipx. pipx itself installed successfully.
`python3 -m pipx ensurepath` also worked great. After this command, I ran a new terminal as suggested. 
`pipx install eth-brownie` however does not work. gives the following error message:
![image](https://user-images.githubusercontent.com/65351091/137639344-df16d180-8089-4a71-834a-e8a6ee790861.png)

Thank you in advance!

**Comment by cromewar** [comment, marked as answer]
I highly recommend install eth-brownie in a virtualenv, try doing this:

`pip3 install virtualenv`

Then create a new virtual environment

`virutalenv nameyouwant`

Activate it (windows)

`.\FolderName\Scripts\activate`

finally install eth-brownie using pip:

 `pip install eth-brownie`

This adds more steps that's true, however gives you more control about which libraries you want depending on the project you are on.

Note: do not forget to install ganache-cli (globally or for the env) using pip.
Note2: I also recommend a lot you try to do this on WSL2 (windows subsystem for linux)

I hope this info helps.

**Reply by haraslub** [reply]
Hello @cromewar,
I am facing the same problem as @s1n7, but your solution does not work for me. I would like to ask you for a help. 
In the step number three, you are referring to activation:

`.\FolderName\Scripts\activate`

(I guess the FolderName is the "nameyouwant" from the step number two, right?)
It does not work for me. Instead I get:

`bash: ./venv/scripts/activate: No such file or directory`

So I tried this:

`sudo ./venv/bin/activate`

but it still does not work, unfortunately. I got:

`sudo: ./venv/bin/activate: command not found`

Any idea what am I suppose to do, please?
I am running Linux Debian 64 bit in Virtual Box in Windows 10.

**Reply by cromewar** [reply]
Hello @haraslub I gave those instructions as it was a windows installation, on linux  is easier, use this command:

`source venv/bin/activate` 

this is all you need.

**Reply by haraslub** [reply]
Hello, thank you for a quick answer. 
It worked! Awesome, thank you very much!!
